python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+            obsLat = visitInfo.observatory.getLatitude().asDegrees()
+            obsElev = visitInfo.observatory.getElevation()
+            earthLocation = astropy.coordinates.EarthLocation.from_geodetic(obsLon, obsLat, obsElev)
+            observatory_gcrs = earthLocation.get_gcrs(astropy.time.Time(obsMJD, format='mjd'))
+            observatory_icrs = observatory_gcrs.transform_to(astropy.coordinates.ICRS())
+            # We want the position in AU in Cartesian coordinates
+            observatories.append(observatory_icrs.cartesian.xyz.to(u.AU).value)
+
+            for row in visitSummary:
+                detector = row['id']
+                if detector not in self.detectors:
+                    self.detectors.append(detector)
+                    detectorBounds = wcsfit.Bounds(row['bbox_min_x'], row['bbox_max_x'],
+                                                   row['bbox_min_y'], row['bbox_max_y'])
+                    instrument.addDevice(str(detector), detectorBounds)
+
+                detectorIndex = np.flatnonzero(detector == np.array(self.detectors))[0]
+                extensionVisits.append(v)
+                extensionDetectors.append(detectorIndex)
+                visits.append(visit)
+                detectors.append(detector)
+                extensionType.append('SCIENCE')
+
+                wcs = row.getWcs()
+                wcss.append(getWCSfromSIP(wcs))
+
+        fieldNumbers = list(np.ones(len(exposureNames), dtype=int) * fieldNumber)
+        instrumentNumbers = list(np.ones(len(exposureNames), dtype=int) * instrumentNumber)
+
+        # Set the reference epoch to be the median of the science visits.
+        # The reference catalog will be shifted to this date.
+        medianEpoch = astropy.time.Time(np.median(mjds), format='mjd').decimalyear
+
+        # Add information for the reference catalog. Most of the values are
+        # not used.
+        exposureNames.append("REFERENCE")
+        self.visits.append(-1)
+        fieldNumbers.append(0)
+        if self.config.useProperMotion:
+            instrumentNumbers.append(-2)
+        else:
+            instrumentNumbers.append(-1)
+        self.ras.append(0.0)
+        self.decs.append(0.0)
+        airmasses.append(0.0)
+        exposureTimes.append(0)
+        mjds.append((refEpoch if (refEpoch is not None) else medianEpoch))
+        observatories.append(np.array([0, 0, 0]))
+        identity = wcsfit.IdentityMap()
+        icrs = wcsfit.SphericalICRS()
+        refWcs = wcsfit.Wcs(identity, icrs, "Identity", np.pi / 180.)
+        wcss.append(refWcs)
+
+        extensionVisits.append(len(exposureNames) - 1)
+        extensionDetectors.append(-1)  # REFERENCE device must be -1
+        visits.append(-1)
+        detectors.append(-1)
+        extensionType.append('REFERENCE')
+
+        # Make a table of information to use elsewhere in the class
+        self.extensionInfo = pd.DataFrame({'visit': visits, 'detector': detectors,
```

Comment:
Does this need to be a dataframe, or could it be a dict of numpy arrays/lists (in the case of the wcss).

Reply:
Yeah, I guess I didn't end up using any dataframe-specific functionality.